src/force-app/main/default/classes/AccountService.cls

```diff
@@ -0,0 +1,40 @@
+public with sharing class AccountService {
+    
+    @TestVisible
+    private static AccountService instance;   
+
+    public static AccountService getInstance() {
+        if (instance == null) {
+            instance = new AccountService();
+        }
+        return instance;
+    }
+
+    public void onBeforeInsert(List<Account> newRecords) {
+        onBeforeInsertActivateStatus(newRecords);
+    }
+
+    public void onBeforeUpdate(List<Account> newRecords, Map<Id,Account> oldRecords) {
+        preventInvalidAccountUpdate(newRecords, oldRecords);
+    }
```

Comment:
What's the reason of having handlers at Service? It should contain business logic only.

30 mins to fix all occurences.